lib/note-content-editor.tsx

```diff
@@ -146,16 +145,17 @@ class NoteContentEditor extends Component<Props> {
 
     const editor = noteChanged ? (goFast ? 'fast' : 'full') : state.editor;
 
+    // reset search selection if the search or note has changed
```

Comment:
adding a little more here might help us avoid wondering again in the future

```ts
// reset search selection when either the note or the search changes
// to avoid, for example, opening a note and having the fourth match selected
```